providers/whispercpp.py

```diff
@@ -0,0 +1,103 @@
+import requests
+import json
+import os
+import subprocess
+import time
+from api.enums import WingmanInitializationErrorType
+from api.interface import WhispercppSttConfig, WhispercppTranscript, WingmanInitializationError
+from os import path
+
+class Whispercpp:
+    def __init__(self):
+        self.base_url: str = ""
+        self.autostart: bool = False
+        self.whispercpp_exe_path: str | None = None
+        self.whispercpp_model_path: str | None = None
+        self.temperature: float = 0.0
+        self.language: str = ""
+        self.times_checked_whispercpp: int = 0
+        
+    def validate_config(self, config: WhispercppSttConfig, errors: list[WingmanInitializationError]):
+        if not errors:
+            errors = []
+
+        if not config.base_url:
+            errors.append(
+                WingmanInitializationError(
+                    wingman_name=self.wingman_name,
+                    message="Missing base url for whispercpp. Please provide a url where whispercpp can be reached.  By default this is http://127.0.0.1:8080.",
+                    error_type=WingmanInitializationErrorType.INVALID_CONFIG,
+                )
+            )
+
+        if config.autostart == True and (config.autostart_settings.whispercpp_exe_path == None or config.autostart_settings.whispercpp_model_path == None or not os.path.exists(config.autostart_settings.whispercpp_exe_path) or not os.path.exists(config.autostart_settings.whispercpp_model_path)):
```

Comment:
If we make `autostart_settings` optional, you have to check if it exists here before accessing the nested properties. You can actually leave out the checks for exe_path and model_path because if these are NOT optional, Pydantic has already validated that they are set for you. That's the beauty of Pydantic and strict typing.